compiler/rustc_session/src/options.rs

```diff
@@ -1502,7 +1502,8 @@ options! {
     incremental: Option<String> = (None, parse_opt_string, [UNTRACKED],
         "enable incremental compilation"),
     inline_threshold: Option<u32> = (None, parse_opt_number, [TRACKED],
-        "set the threshold for inlining a function"),
+        "this option is deprecated and does nothing \
+        (consider using `-Cllvm-args=--inline-threshold=...`)"),
```

Comment:
Will this have any effect if they do?

Reply:
Yes that makes a difference. Building libstd with `-Cllvm-args=--inline-threshold=100000` vs `-Cllvm-args=--inline-threshold=2` reliably makes a big difference in the size of `.text`, namely 1594 kB vs 367 kB:

```console
$ RUSTFLAGS="-Ccodegen-units=1 -Cllvm-args=--inline-threshold=100000" ./x build --stage 0 library/std
$ size ./build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/release/libstd.rlib
   text   ...
      0   ...
1594437   ...
$ RUSTFLAGS="-Ccodegen-units=1 -Cllvm-args=--inline-threshold=2" ./x build --stage 0 library/std
$ size ./build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/release/libstd.rlib
   text   ...
      0   ...
 367682   ...
```

Reply:
Oh neat, I'm surprised it does anything!